ledger/participant-state/kvutils/tools/integrity-check/src/main/scala/ledger/participant/state/kvutils/tools/integritycheck/RawWriteSetComparison.scala

```diff
@@ -54,15 +58,18 @@ final class RawWriteSetComparison(stateKeySerializationStrategy: StateKeySeriali
     val differencesExplained = expectedWriteSet
       .zip(actualWriteSet)
       .map { case ((expectedKey, expectedValue), (actualKey, actualValue)) =>
-        if (expectedKey == actualKey && expectedValue != actualValue) {
+        // We need to compare `.bytes` because the type will be different.
+        // Unfortunately, the export format doesn't differentiate between log and state entries.
```

Comment:
The reason for that is that on a real ledger (unless we store these separately) we cannot distinguish them either.